src/app/services/variable.service.ts

```diff
@@ -67,11 +67,8 @@ export class VariableService {
     Observable.combineLatest(
       this.countService.selectedConceptCountMapUpdated.asObservable(),
       this.treeNodeService.treeNodesUpdated.asObservable()
-    ).subscribe(res => {
-      const isTreeLoadingFinished = res[1];
-      if (isTreeLoadingFinished) {
-        this.updateVariables();
-      }
```

Comment:
should wait for tree finishing loading before updating the variables, right?

Reply:
but the res[1] from combineLatest subscription was not always a tree ( depending on what was emitted as the latest, it was sometimes a result of the count map update). Also it looks like waiting for the tree nodes to be loaded is already a part of `updateVariables`

Reply:
res[1] should return a boolean value indicating if the tree is finished loading, but you are right, `updateVariables` already takes care of the tree loading waiting, so we can remove the res[1] checking :)